aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -25,10 +26,21 @@ void bitwise_not_kernel_cuda(TensorIterator& iter) {
   }
 }
 
+template<typename scalar_t>
+__host__ __device__ static inline scalar_t exp_wrapper(scalar_t v) {
+  return ::exp(v);
+}
+
+template<typename T>
+__host__ __device__ static inline c10::complex<T> exp_wrapper(c10::complex<T> v) {
+  return c10_complex_math::exp(v);
+}
+
 void exp_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "exp_cuda", [&]() {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(ScalarType::Half, iter.dtype(), "exp_cuda", [&]() {
     gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
-      return ::exp(a);
+      // We do not use std::exp directly here because ROCm doesn't work well with std:: functions
+      return exp_wrapper(a);
```

Comment:
I believe you can now use `::exp` here, and remove the wrapper

Reply:
Thanks. Could you point me to the definition of the global function?

Reply:
https://github.com/pytorch/pytorch/blob/master/c10/util/complex_math.h#L218